bin/prefix-vendor-namespace.php

```diff
@@ -57,39 +60,50 @@
 			continue 2;
 		}
 
+		$namespace_change = 0;
+		$uses_change      = 0;
+
 		$contents =	preg_replace(
 			"#^(\s*)(namespace)\s*({$quoted}[\\\\|;])#m",
 			"\$1\$2 {$new_namespace}\\\\\$3",
-			$contents
+			$contents,
+			-1,
+			$namespace_change
 		);
 
 		$contents =	preg_replace(
 			"#^(\s*)(use)\s*({$quoted}\\\\)#m",
 			"\$1\$2 {$new_namespace}\\\\\$3",
-			$contents
+			$contents,
+			-1,
+			$uses_change
 		);
 
 		if ( ! empty( $direct_replacements[ $path ] ) ) {
 			foreach ( $direct_replacements[ $path ] as $replace ) {
-				$replace  = preg_quote( $replace, '#' );
-				$contents =	preg_replace(
+				$direct_change = 0;
+				$replace       = preg_quote( $replace, '#' );
+				$contents      = preg_replace(
 					"#({$replace})#m",
 					"{$new_namespace}\\\\\$1",
-					$contents
+					$contents,
+					-1,
+					$direct_change
 				);
+				if ( $direct_change ) {
+					$uses_change += $direct_change;
+				}
 			}
 		}
+		if ( ! $namespace_change && $uses_change ) {
+			$file_notices[] = $file;
+		}
 
 		file_put_contents( $file, $contents );
 	}
 
-	// Update the namespace in the composer.json files.
-	$composer_files = array_filter(
-		explode(
-			"\n",
-			`find "{$vendor_dir}/{$path}" -iname 'composer.json'`
-		)
-	);
+	// Update the namespace in the composer.json files, recursively finding all files named explicitly "composer.json".
+	$composer_files = get_dir_contents($vendor_dir . '/' . $path, '/' . preg_quote(DIRECTORY_SEPARATOR . 'composer.json') . '$/');
```

Comment:
This line doesn't work on platforms where DIRECTORY_SEPARATOR is `/`. That's because the default deliminator for preg_quote is null. So unless you pass the deliminator which is being used in the regex `/` then it doesn't escape it.

It's building a regex like this:
```
//composer.json$/
```

As you can see this causes all the automated tests to fail because it thinks the `c` is a regex modifier.

```suggestion
	$composer_files = get_dir_contents(
		"{$vendor_dir}/{$path}",
		'/' . preg_quote( DIRECTORY_SEPARATOR, '/' ) . 'composer.json$/'
	);
```